inc/api.class.php

```diff
@@ -125,16 +125,28 @@ public function initApi() {
 
       // check ip access
       $apiclient = new APIClient;
-      $where_ip = "";
+      $where_ip = [];
       if ($this->ipnum) {
-         $where_ip .= " AND (`ipv4_range_start` IS NULL
-                             OR (`ipv4_range_start` <= '$this->ipnum'
-                                 AND `ipv4_range_end` >= '$this->ipnum'))";
+         $where_ip = [
+            'OR' => [
```

Comment:
I'm not sure about the `OR` here with the else part

Reply:
Currently:
```php
$where_ip = [
   'OR' => [
      'ipv4_range_start' => null,
      [
         'AND' => [
            'ipv4_range_start'   => ['<=', 10],
            'ipv4_range_end'     => ['>=', 10]
         ]
      ]
   ]
];

$crit = [
   'FROM'   => 'table',
   'WHERE'  => [
      ['is_active' => 1] + $where_ip
   ]
];

$iterator->buildQuery($crit);
echo $iterator->getSql();
```

Will give:
```sql
SELECT
    *
FROM
    `table`
WHERE
    (
        `is_active` = '1' AND(
            `ipv4_range_start` IS NULL OR(
                (
                    `ipv4_range_start` <= '10' AND `ipv4_range_end` >= '10'
                )
            )
        )
    )
```

Reply:
so this is wrong :D